sphinx/builders/linkcheck.py

```diff
@@ -345,7 +357,7 @@ class CheckResult(NamedTuple):
     uri: str
     docname: str
     lineno: int
-    status: _StatusUnknown
+    status: _Status | None
```

Comment:
It'd be nice to remove the possibility to allow `None` here, even as a type-hint; really a `CheckResult` should always have a status.

Reply:
Perhaps best to handle this in a subsequent PR that also removes the possibility of `retries=0`.

Reply:
No, let's use the `unknown` status here.  Again, in practice it _shouldn't_ really happen, but let's begin by using that placeholder value and then subsequently refactor to ensure that it can't happen (and remove it).